### DISCUSSION TITLE
Support for ratings

### DISCUSSION THREAD

**Discussion by jfgoncalves** [ryck/scrobblex#2] [Ideas]
I was wondering if it would be possible to add support for ratings (episode, season, show, movie)?
I know Plex exposes the event in its webhooks with media.rate.

I was looking for something more lightweight than PlexTraktSync.

**Comment by ryck** [comment]
It's a really good idea! I'll have a look, it shouldn't be that difficult (famous last words...)

This will only push the ratings to trakt when you do it in plex, syncing is not in scope, so it's always going to be simpler than PlexTraktSync :)

**Reply by jfgoncalves** [reply]
Yup that's fine and totally what I'm looking for. Thanks for looking into it!

**Reply by ryck** [reply]
Give it a try, rates are supported in v0.7.0!

**Comment by jfgoncalves** [comment]
After doing some testing yesterday I can confirm that the ratings are working very well on all type of media! The scrobble also works but I feel like it struggles to pick up the events like play pause and stop and it adds a little delay.
I've actually had more trouble with the behind the scenes setup.

On my setup:
- I have put Scrobblex behind a reverse proxy (nginx proxy manager) with SSL certificate.
- The reverse proxy only resolves for machines on my local network (as the domain DNS records points to the local IP of my reverse proxy), with Plex running in it.
- I run Scrobblex using the provided docker compose.
- I run Plex using linuxserver's image.
- Plex is exposed to the Internet.
- Scrobblex is NOT exposed to the internet.

On the app:
- If I use https in the redirect uri of the trakt app, Authorising (I made sure to access the app via https first) would return a trakt error page with "malformed response". So I use http there.
- If I force SSL via the reverse proxy, the token never get retrieved even when using http in the above redirect uri.
- If I use https in the Plex Webhook field, it works but returns a 408 regardless of using http or https.
- If I use http with the local IP:port of the app container, it works but also return a 408.

Is it expected behaviour for the app to return a 408 to the Plex console? I was expecting a 200. I want to point that even if it returns a 408 it still works, the item gets matched against trakt and action is taken.
Should I expose Scrobblex to the internet for it to give a 200? Do you need me to provide logs?

**Comment by ryck** [comment]
Yeah, rates events are more consistent too in my experience... play ones, not so much, especially the `scrobble` one, but not much I can do there... at least I am also sending every pause / resume event, which helps a little...

Regarding setup, I haven't had any issue and I never got a 408, so not really sure what's going on.
The app is a simple nodejs app, so it doesn't need much, only to be on the same network than plex so the event can reach scrobblex...
It doesn't event need to be exposed to the internet, because you can add localhost in trakt as the callback (mine is not exposed, for example)

I'd check if both container can see each other and take it from there...

**Reply by ryck** [reply]
Btw, I am now just using the scrobble event (unless you are on debug mode) so that should help to not confuse the Trakt API and improve the scrobbling
